src/utils.ts

```diff
@@ -201,6 +201,13 @@ export namespace Utils {
       }
     }
 
+    // verify homeDomain
+    if (homeDomain && `${homeDomain} auth` !== operation.name) {
```

Comment:
See the Go SDK, which does check equality with the 2nd `auth` token in the string: https://github.com/stellar/go/pull/3182/files#diff-bd94368e661d629d7679f3b3eb9da1f18e91ca7fc31a3f418f53109677df631aR959

Maybe this is fine too, but I think the Go SDK approach is slightly better because it's a bit more forgiving while still enforcing that the home domain match, which is the important piece.

Reply:
✅  I think what's here in this PR is fine and probably more correct. The specification says the field is `{home domain} auth`. The Go change should probably be changed to match.